src/ui/purchases-ui.svelte

```diff
@@ -215,22 +216,27 @@
 
     priceBreakdown.taxCalculationStatus = "loading";
 
-    const taxCalculation = await purchaseOperationHelper.checkoutCalculateTax(
-      taxCustomerDetails?.countryCode,
-      taxCustomerDetails?.postalCode,
-    );
-
-    priceBreakdown.taxCalculationStatus = "calculated";
-    priceBreakdown.totalAmountInMicros = taxCalculation.total_amount_in_micros;
-    priceBreakdown.taxAmountInMicros = taxCalculation.tax_amount_in_micros;
-    priceBreakdown.totalExcludingTaxInMicros =
-      taxCalculation.total_excluding_tax_in_micros;
-    priceBreakdown.taxBreakdown =
-      taxCalculation.pricing_phases.base.tax_breakdown;
-    priceBreakdown.pendingReason = null;
-
-    gatewayParams.elements_configuration =
-      taxCalculation.gateway_params.elements_configuration;
+    try {
+      const taxCalculation = await purchaseOperationHelper.checkoutCalculateTax(
+        taxCustomerDetails?.countryCode,
+        taxCustomerDetails?.postalCode,
+      );
+      priceBreakdown.taxCalculationStatus = "calculated";
+      priceBreakdown.totalAmountInMicros =
+        taxCalculation.total_amount_in_micros;
+      priceBreakdown.taxAmountInMicros = taxCalculation.tax_amount_in_micros;
+      priceBreakdown.totalExcludingTaxInMicros =
+        taxCalculation.total_excluding_tax_in_micros;
+      priceBreakdown.taxBreakdown =
+        taxCalculation.pricing_phases.base.tax_breakdown;
+      priceBreakdown.pendingReason = null;
+      gatewayParams.elements_configuration =
+        taxCalculation.gateway_params.elements_configuration;
+    } catch (e) {
+      priceBreakdown.taxCalculationStatus = "pending";
+      priceBreakdown.pendingReason = "needs_complete_billing_address";
+      priceBreakdown.taxCalculationBasedOnFullAddress = true;
```

Comment:
why do we need this flag if we already have a specific `pending_reason`?

Reply:
We need the flag since when the address is resolved correctly the status moves to "complete" and the pendingReason becomes null.

If we don't have the flag the address elements will be hidden.

Reply:
Got it, thanks for the explanation!